src/core/thread/mesh_forwarder.cpp

```diff
@@ -460,6 +466,16 @@ otError MeshForwarder::HandleFrameRequest(Mac::TxFrame &aFrame)
         SendMesh(*mSendMessage, aFrame);
         break;
 
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+    case Message::kTypeMacEmptyData:
+    {
+        Mac::Address macDestAddr;
+        macDestAddr.SetShort(Get<Mle::MleRouter>().GetParent().GetRloc16());
+        Get<IndirectSender>().PrepareEmptyFrame(aFrame, macDestAddr, /* aAckRequest */ false);
+    }
+    break;
+#endif
```

Comment:
Move out of `#if OPENTHREAD_FTD` scope?

Reply:
should this unicast packet request ack?

Reply:
Note that we are now reusing the code from `IndirectSender`which only exists in FTD.